rewriter/Command.cc

```diff
@@ -84,9 +86,24 @@ void Command::run(core::MutableContext ctx, ast::ClassDef *klass) {
     ast::MethodDef::Flags flags;
     flags.isSelfMethod = true;
     flags.discardDef = true;
-    auto selfCall = ast::MK::SyntheticMethod(call->loc, call->loc, call->name, std::move(newArgs),
+    auto selfCall = ast::MK::SyntheticMethod(call->loc, call->declLoc, call->name, std::move(newArgs),
                                              ast::MK::RaiseTypedUnimplemented(call->declLoc), flags);
 
+    // We are now in the weird situation where we have an actual method that
+    // the user has written, but we have a synthetic method that lives at the
+    // same location.  If we try to jump-to-def from the actual method, there
+    // are no calls to it, which will frustrate the user.  Erase the location(s)
+    // on the non-synthetic method so that LSP only sees the synthetic method.
```

Comment:
Did you consider an alternative solution which ordered the synthetic and non synthetic methods in the reverse order, so that the LSPQuery::byLoc has the synthetic one at index 0 instead?

Reply:
We tried the solution where we flipped the ordering of the `call` definitions offline and it did not work, so we're going with the approach in the PR.